pkg/apis/contrail/v1alpha1/vrouter_types.go

```diff
@@ -134,6 +134,19 @@ func (c *Vrouter) PrepareDaemonSet(ds *appsv1.DaemonSet,
 		instanceType: request.Name}
 	ds.Spec.Template.SetLabels(map[string]string{"contrail_manager": instanceType,
 		instanceType: request.Name})
+	ds.Spec.Template.Spec.Affinity = &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{
+				LabelSelector: &metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{{
+						Key:      instanceType,
+						Operator: "Exists",
+					}},
+				},
+				TopologyKey: "kubernetes.io/hostname",
+			}},
+		},
+	}
```

Comment:
Does it make sense since DaemonSet runs single replica on every node?

Reply:
Yes because on Openshift we have 2 vrouter daemonsets. Have a look at the explanation in description.